src/App.vue

```diff
@@ -13,21 +13,44 @@ export default Vue.extend( {
 	data() {
 		return {
 			isi18nLoaded: false as boolean,
+			lang: 'en' as string,
 		};
 	},
-	beforeCreate(): void {
+	created(): void {
 		const fetchi18n = async (): Promise<void> => {
-			const response = await fetch( 'i18n/en.json' );
-			const messages = {
-				en: await response.json(),
+			const urlParams = new URLSearchParams( window.location.search );
+			if ( urlParams.has( 'uselang' ) ) {
+				this.lang = urlParams.get( 'uselang' ) as string;
+			}
+
+			const responseEn = await fetch( 'i18n/en.json' );
+			const messages: { [key: string]: { [key: string]: string} } = {
+				en: await responseEn.json(),
 			};
+
+			if ( this.lang !== 'en' ) {
+				const responseLang = await fetch( 'i18n/' + this.lang + '.json' ).catch(
+					() => {
+						// TODO: Show a user-facing notification instead
+						console.warn( 'The language requested could not be retrieved, falling back to English' );
+						messages[ this.lang ] = {};
+					},
+				) as Response;
+				if ( responseLang.ok ) {
```

Comment:
Are we sure that `responseLang` is defined if the catch above is triggered?

Reply:
yeah it needs clean up